tests/integration_tests/vectorstores/test_pinecone.py

```diff
@@ -132,28 +146,68 @@ def test_from_texts_with_scores(self, embedding_openai: OpenAIEmbeddings) -> Non
         ]
         assert scores[0] > scores[1] > scores[2]
 
+    def test_from_existing_index_with_namespaces(
+        self, embedding_openai: OpenAIEmbeddings
+    ) -> None:
+        """Test that namespaces are properly handled."""
+        # Create two indexes with the same name but different namespaces
+        texts_1 = ["foo", "bar", "baz"]
+        metadatas = [{"page": i} for i in range(len(texts_1))]
+        Pinecone.from_texts(
+            texts_1,
+            embedding_openai,
+            index_name=index_name,
+            metadatas=metadatas,
+            namespace=f"{index_name}-1",
+        )
+
+        texts_2 = ["foo2", "bar2", "baz2"]
+        metadatas = [{"page": i} for i in range(len(texts_2))]
+
+        Pinecone.from_texts(
+            texts_2,
+            embedding_openai,
+            index_name=index_name,
+            metadatas=metadatas,
+            namespace=f"{index_name}-2",
+        )
+
+        # Search with namespace
+        docsearch = Pinecone.from_existing_index(
+            index_name=index_name,
+            embedding=embedding_openai,
+            namespace=f"{index_name}-1",
+        )
+        output = docsearch.similarity_search("foo", k=20, namespace=f"{index_name}-1")
+        # check that we don't get results from the other namespace
+        page_contents = sorted(set([o.page_content for o in output]))
+        assert all(content in ["foo", "bar", "baz"] for content in page_contents)
+        assert all(content not in ["foo2", "bar2", "baz2"] for content in page_contents)
+
     def test_add_documents_with_ids(
         self, texts: List[str], embedding_openai: OpenAIEmbeddings
     ) -> None:
         ids = [uuid.uuid4().hex for _ in range(len(texts))]
         Pinecone.from_texts(
-            texts=texts, ids=ids, embedding=embedding_openai, index_name=index_name
+            texts=texts,
+            ids=ids,
+            embedding=embedding_openai,
+            index_name=index_name,
+            namespace=index_name,
         )
-        # wait for the index to be ready
-        time.sleep(20)
         index_stats = self.index.describe_index_stats()
-        assert index_stats["total_vector_count"] == len(texts)
+        assert index_stats["namespaces"][index_name]["vector_count"] == len(texts)
 
         ids_1 = [uuid.uuid4().hex for _ in range(len(texts))]
         Pinecone.from_texts(
             texts=texts,
             ids=ids_1,
             embedding=embedding_openai,
             index_name=index_name,
+            namespace=index_name,
         )
-        # wait for the index to be ready
-        time.sleep(20)
         index_stats = self.index.describe_index_stats()
+        assert index_stats["namespaces"][index_name]["vector_count"] == len(texts) * 2
         assert index_stats["total_vector_count"] == len(texts) * 2
```

Comment:
I think can remove this line as was added in prev PR?

Reply:
this is from sep pr than one being reverted afaict